e2e/app/component/share-modal.ts

```diff
@@ -0,0 +1,107 @@
+import Container from 'app/container';
+import {ElementHandle, Page} from 'puppeteer';
+import {WorkspaceAccessLevel} from 'app/page-identifiers';
+import {ElementType} from 'app/xpath-options';
+import Button from 'app/element/button';
+import Textbox from 'app/element/textbox';
+import ClrIcon from 'app/element/clr-icon-link';
+
+
+export enum ButtonLabel {
+  Cancel = 'Cancel',
+  Save = 'Save'
+}
+
+const Selector = {
+  defaultDialog: '//*[@role="dialog"]',
+}
+
+export default class ShareModal extends Container {
+
+  constructor(page: Page, xpath: string = Selector.defaultDialog) {
+    super(page, xpath);
+  }
+
+  async shareWithUser(username: string, level: WorkspaceAccessLevel): Promise<void> {
+    const searchBox = await this.waitForSearchBox();
+    await searchBox.type(username);
+
+    const addCollab = await this.waitForAddCollaboratorIcon();
+    await addCollab.click();
+
+    const roleInput = await this.waitForRoleSelectorForUser(username);
+    await roleInput.click();
+
+    const ownerOpt = await this.waitForRoleOption(level);
+    await ownerOpt.click();
+
+    await this.clickButton(ButtonLabel.Save);
+    await this.waitUntilDialogIsClosed();
+  }
+
+  async removeUser(username: string): Promise<void> {
+    const rmCollab = await this.page.waitForXPath(
+      `${this.collabRowXPath(username)}//clr-icon[@shape="minus-circle"]`);
+    await rmCollab.click();
+
+    await this.clickButton(ButtonLabel.Save);
+    await this.waitUntilDialogIsClosed();
+    return;
+  }
+
+  /**
+   * Creates an xpath to a share "row" for a given user within the modal. This
+   * can be combined with a child selector to pull out a control for a user.
+   */
+  private collabRowXPath(username: string): string {
+    // We dip into child contents to find the collab user row element parent.
+    // .//div filters by a relative path to the parent row.
+    return `//*[@data-test-id="collab-user-row" and .//div[` +
```

Comment:
nit: If this class extends `Dialog`, you can also adds `${this.getXpath()}` at the beginning of xpath to ensure element is located in dialog.

Reply:
Done